### PR TITLE
[v7r0] makde dirac-install python3 compatible

### PR DESCRIPTION
**Pull request by zmathe**
BEGINRELEASENOTES

*Core
FIX: Make dirac-install tool python 3 compatible

ENDRELEASENOTES

### CODE REVIEW

File: .github/workflows/pilot.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: dirac-install
```
Comment (fstagni) [inline .github/workflows/pilot.yml]: Change name of this workflow from "pilot.yml" to diracinstall.yml

File: Core/Utilities/Platform.py [hunk 2]
```diff
@@ -2,25 +2,29 @@
 Compile the externals
 """
 
-__RCSID__ = "$Id$"
+from __future__ import unicode_literals
+from __future__ import absolute_import, division, print_function
```
Comment (fstagni) [inline Core/Utilities/Platform.py]: One one line only.

**Comment by fstagni** [comment]
Check https://github.com/DIRACGrid/DIRAC/pull/4411/checks?check_run_id=393508333#step:3:42 (and why it doesn't fail)

**Comment by zmathe** [comment]
:D Ah yes, It fail because I also fixed dirac-install ...  The exception is caught, that's why does not fail.

**Comment by zmathe** [comment]
@fstagni It should be fine now.

File: .github/workflows/diracinstall.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    - name: prepare environment
+      run: |
+        conda config --set add_pip_as_python_dependency false
+        conda create -c conda-forge -c free -n python_${{ matrix.python }} python=${{ matrix.python }}
+        # curl -O https://raw.githubusercontent.com/DIRACGrid/DIRAC/integration/Core/scripts/dirac-install.py
+    - name: run dirac-install script
+      run: |
+        eval "$(conda shell.bash hook)" && conda activate python_${{ matrix.python }}
+        # python dirac-install.py -l DIRAC -r integration -t server  --dirac-os -m https://github.com/zmathe/DIRAC.git:::DIRAC:::pcompatibility -ddd
+        python Core/scripts/dirac-install.py -l DIRAC -r integration -t server  --dirac-os -m https://github.com/zmathe/DIRAC.git:::DIRAC:::pcompatibility -ddd
```
Comment (fstagni) [inline .github/workflows/diracinstall.yml]: I guess this version is just for your tests

File: .github/workflows/diracinstall.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      #  ref: pcompatibility
+    - name: prepare environment
+      run: |
+        conda config --set add_pip_as_python_dependency false
+        conda create -c conda-forge -c free -n python_${{ matrix.python }} python=${{ matrix.python }}
+        # curl -O https://raw.githubusercontent.com/DIRACGrid/DIRAC/integration/Core/scripts/dirac-install.py
+    - name: run dirac-install script
+      run: |
+        eval "$(conda shell.bash hook)" && conda activate python_${{ matrix.python }}
+        # python dirac-install.py -l DIRAC -r integration -t server  --dirac-os -m https://github.com/zmathe/DIRAC.git:::DIRAC:::pcompatibility -ddd
```
Comment (fstagni) [inline .github/workflows/diracinstall.yml]: you can remove the comments

File: .github/workflows/diracinstall.yml (hunk 3 above)
Comment (fstagni) [reply .github/workflows/diracinstall.yml]: BTW I have added "master" in https://github.com/DIRACGrid/DIRAC/pull/4377 which is probably what we want to use here.

File: Core/Utilities/Platform.py [hunk 5]
```diff
@@ -35,14 +38,14 @@ def libc_ver( executable = sys.executable, lib = '', version = '',
...
+  with open(executable, 'rb') as f:
+    binary = f.read(chunksize)
     pos = 0
     version = [0, 0, 0]
     while True:
-      m = _libc_search.search( binary, pos )
+      m = _libc_search.search(str(binary), pos)
       if not m:
-        binary = f.read( chunksize )
+        binary = f.read(chunksize)
```
[the remark this answers is not in the document]
Comment (zmathe) [reply Core/Utilities/Platform.py]: I applied your proposal. I think you are right, but it is depend on the file size. If f.read(chunksize) can not read all bytes from the file, it never match.

File: Core/Utilities/Platform.py [hunk 6]
```diff
@@ -68,18 +71,18 @@ def libc_ver( executable = sys.executable, lib = '', version = '',
...
         if lib != 'glibc':
           lib = 'libc'
-          version = max( version, soversion )
-          if threads and version[-len( threads ):] != threads:
+          version = max(version, [int(soversion)]).pop()
+          if threads and version[-len(threads):] != threads:
             version = version + threads
       pos = m.end()
-  return lib, '.'.join( map( str, version ) )
+  return lib, '.'.join(map(str, version))
```
Comment (chrisburr) [inline Core/Utilities/Platform.py]:
```suggestion
  return lib.decode(), '.'.join(map(str, version))
```
Also, I think the 5 strings containing `'glibc'`/`'libc'` should be bytes (`b'glibc'`/`b'libc'`) else they will never be equal in the comparisons.

File: Core/Utilities/Platform.py [hunk 7]
```diff
@@ -88,63 +91,66 @@ def getPlatformString():
       spStdout = sp.stdout
     except:
```
Comment (chrisburr) [inline Core/Utilities/Platform.py]:
```suggestion
    except Exception:
```

**Pull request merged**